src/ucs/sys/sock.c

```diff
@@ -332,18 +354,54 @@ ucs_socket_handle_io_error(int fd, const char *name, ssize_t io_retval, int io_e
     return status;
 }
 
+/**
+ * Handle the IO operation if it was successful.
+ *
+ * @param [in]  fd         The socket fd.
+ * @param [in]  data       The pointer to user's data or pointer to the array of
+ *                         iov elements.
+ * @param [in]  count      The length of user's data or the number of elemnts in
+ *                         the array of iov.
+ * @param [out] length_p   Pointer to the result length of user's data that was
+ *                         sent/received.
+ * @param [in]  io_retval  The result of the IO operation.
+ * @param [in]  is_iov     Flag that specifies type of the operation (1 if vector
+ *                         operation).
+ *
+ * @return if the IO operation was successful - 1, otherwise - 0.
+ */
+static inline int
+ucs_socket_handle_io_success(int fd, const void *data, size_t count,
```

Comment:
maybe unite ucs_socket_handle_io_success() and ucs_socket_check_errno() to one function. it will accept all information and return ucs_status_t to indicate success/failure.